examples/basic/kubernetes-hub/README.md

```diff
@@ -1,19 +1,246 @@
+        topology.WithKubernetesConfig(""),
+        topology.WithKubernetesCRI("/run/containerd/containerd.sock"),
+        topology.WithKubernetesNamespace(""),
+        topology.WithKubernetesProcRoot("/"),
+        topology.WithKubernetesLabelSelector("noSwoll!=true"),
+        topology.WithKubernetesFieldSelector("status.phase=Running"),
+    )
+    if err != nil {
+        log.Fatalf("Could not create the kubernetes observer: %v", err)
+    }
+```
+
+Here we have created the first part of a `Topology`, the `Observer`. In thise case, the `Observer` will interact with Kubernetes and the CRI (Container Runtime Interface) to map numerical kernel-namespaces to a specific (k8s)Namespace/POD/container. There are a few `With` configuration directives set here, so let's go over each one individually:
+
+```go
+topology.WithKubernetesConfig("")
+```
+
+If this code was being run outside of the cluster (e.g., not an `in-cluster` deployment), the value of this option would be the fully-qualified path to your current kubernetes configuration file.
+
+```go
+topology.WithKubernetesCRI("/run/containerd/containerd.sock")
+```
+
+This is the path to the CRI socket file as seen by the host running this code. This socket must be readable via a shared mount.
+
+```go
+topology.WithKubernetesNamespace("")
+```
+
+You can tell our observer to limit the container search to a single kubernetes namespace. For the sake of brevity, we leave this empty. The result of which is monitoring container states in *all* namespaces.
+
+```go
+topology.WithKubernetesProcRoot("/")
+```
+
+The Observer uses ProcFS to derive what numerical kernel-namespaces a process belongs to. If that is mounted in a different directory outside of `/`, you would set it here.
+
+```go
+topology.WithKubernetesLabelSelector("noSwoll!=true")
+```
+
+This acts like a pre-filter for container events, essentially stating that any Pod/Container that has the label `noSwoll=true` should never be seen.
+
+```go
+topology.WithKubernetesFieldSelector("status.phase=Running")
+```
+
+Inform the Observer that we are only interested in containers that Kube has deemed as "running".
+
+### Prime & Pump the Hub
+
+```go
+    hub, err := topology.NewHub(bpf, observer)
+    if err != nil {
+        log.Fatalf("Could not create the hub: %v", err)
+    }
+```
+
+Creates a brand new Hub context, using the Kubernetes Observer object we just created.
+
+```go    
+    if err := hub.Probe().DetectAndSetOffsets(); err != nil {
+        log.Fatalf("Could not detect offsets for running kernel: %v", err)
+    }
+```
+
+This part is pretty important. Since our BPF was compiled on a different system, there are various members of the `struct task_struct` structure that we need to read. These structure members live at offsets that are unknown to the system at runtime. This helper function `probe.DetectAndSetOffsets()` will poke at the memory of the running kernel in order to set  the proper offsets.
```

Comment:
Would it be more accurate to say that there are members of task_struct you need to read, and because our BPF was compiled on a different system the offsets may not match what was known at compile time? Just confirming because the sentence currently reads like "because it was compiled on another system, we must read members of the struct".

Reply:
good point. i need to reword some of this anyway - zero sleep at the time of writing.